### PR TITLE
avoid duplicate key (x-ms-retry-after-ms) exception

### PR DESCRIPTION
**Pull request by rajithaalurims**
## Description

When customer posts bundle transactions, sometimes below error is noticed in exception logs.
System.ArgumentException: An item with the same key has already been added. Key: x-ms-retry-after-ms.

I am not able to reproduce this, but based on the behavior of trace logs when compared for failed and successful bundle posts and exception stack trace, the error happens in Microsoft.Health.Fhir.Api.Features.ActionResults.ResourceActionResult class while trying to add headers.

I added a check to remove the key if it already exists before adding it.

## Related issues
Addresses [issue #[95285](https://microsofthealth.visualstudio.com/Health/_boards/board/t/Olympus/Stories/?workitem=95285)].

## Testing
Ran E2E tests (BatchTests and TransactionTests) successfully.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with **Azure Healthcare APIs** if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs [hunk 1]
```diff
@@ -68,6 +68,11 @@ public override Task ExecuteResultAsync(ActionContext context)
 
             foreach (KeyValuePair<string, StringValues> header in Headers)
             {
+                if (response.Headers.ContainsKey(header.Key))
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs:71]: Could we add a unit test to show it properly handles duplicate headers now?

Comment (rajithaalurims) [reply src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs:71]: Done. Please review

File: src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/ActionResults/FhirResultTests.cs [hunk 2]
```diff
@@ -62,5 +68,36 @@ public void GivenANotFoundStatus_WhenReturningAResult_ThenTheStatusCodeIsSetCorr
...
+            ServiceCollection collection = new ServiceCollection();
+            collection.AddSingleton<IActionResultExecutor<ObjectResult>>(executor);
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            collection.AddSingleton<RequestContextAccessor<IFhirRequestContext>>(contextAccessor);
+
+            ServiceProvider provider = collection.BuildServiceProvider();
+            context.HttpContext.RequestServices = provider;
+
+            result.Headers.Add("testKey", "1");
+            context.HttpContext.Response.Headers.Add("testKey", "1");
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/ActionResults/FhirResultTests.cs:93]: Could you change the value of the header and check its value in the assert to show the header in the result is the one posted?

Comment (rajithaalurims) [reply src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/ActionResults/FhirResultTests.cs]: Done. Also added another header variable to verify that value is not being modified.

**Review by LTA-Thinking — APPROVED** [review]
